pkg/assembler/backends/neo4j/cerifyBad.go

```diff
@@ -17,11 +17,294 @@ package neo4jBackend
 
 import (
 	"context"
-	"fmt"
+	"strings"
 
 	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j/dbtype"
+	"github.com/vektah/gqlparser/v2/gqlerror"
 )
 
 func (c *neo4jClient) CertifyBad(ctx context.Context, certifyBadSpec *model.CertifyBadSpec) ([]*model.CertifyBad, error) {
-	panic(fmt.Errorf("not implemented: CertifyBad - CertifyBad"))
+	certifyPkg := false
+	certifySrc := false
+	certifyArt := false
+	queryAll := false
+	if certifyBadSpec.Package != nil {
+		certifyPkg = true
+	}
+	if certifyBadSpec.Source != nil {
+		certifySrc = true
+	}
+	if certifyBadSpec.Artifact != nil {
+		certifyArt = true
+	}
+	if certifyBadSpec.Package == nil && certifyBadSpec.Source == nil && certifyBadSpec.Artifact == nil {
+		queryAll = true
+	}
+
+	if certifyPkg && certifySrc && certifyArt {
+		return nil, gqlerror.Errorf("cannot specify package, source and artifact together for CertifyBad")
+	}
+	if certifyPkg && certifySrc {
+		return nil, gqlerror.Errorf("cannot specify package and source together for CertifyBad")
+	}
+	if certifyPkg && certifyArt {
+		return nil, gqlerror.Errorf("cannot specify package and artifact together for CertifyBad")
+	}
+	if certifySrc && certifyArt {
+		return nil, gqlerror.Errorf("cannot specify source and artifact together for CertifyBad")
+	}
```

Comment:
nit: maybe a bit more readable using a helper that checks for only one boolean and err message that only one should be specified.